src/utils/blockfrost-api.ts

```diff
@@ -30,15 +31,15 @@ export const getBlockData = async (options?: {
   // Fetch latest block and all addresses affected in the block
   // Fetching of affected addresses may fail, there are 3 retry attempts before throwing an error
   const MAX_ATTEMPTS = 3;
-  const latestBlock = options?.block
-    ? await blockfrostAPI.blocks(options.block)
-    : await blockfrostAPI.blocksLatest();
+  const latestBlock = await limiter(() =>
```

Comment:
I worry if slowing down getBlockData couldn't cause some issue as it is rally essential functionality, meaning we can delay clients' requests for `GET_ACCOUNT_INFO` or something, but delaying processing of new blocks/txs for every client may not be desired. 

Anyways, let's try it, I am leaving this comment as a mental note for myself 🤞